x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/ttest/TTestAggregatorFactory.java

```diff
@@ -63,13 +76,46 @@ protected Aggregator doCreateInternal(SearchContext searchContext,
         }
         switch (testType) {
             case PAIRED:
+                if (filterA != null || filterB != null) {
+                    throw new IllegalArgumentException("Paired t-test doesn't support filters");
+                }
                 return new PairedTTestAggregator(name, numericMultiVS, tails, format, searchContext, parent, metadata);
             case HOMOSCEDASTIC:
-                return new UnpairedTTestAggregator(name, numericMultiVS, tails, true, format, searchContext, parent, metadata);
+                return new UnpairedTTestAggregator(name, numericMultiVS, tails, true, this::getWeights, format, searchContext, parent,
```

Comment:
Should we validate/throw an exception/something if people specify an unpaired type, but don't provide filters?  Not sure where to put that validation logic but it seems we should disallow folks from basically running an unpaired test on two match-all's?

Reply:
I was thinking about support for something like this: 
```
{"before": [1,2,3], "after":[6,7,8,9,10]}
```
 It currently works for unpaired, fails on paired and doesn't require filters.